app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java

```diff
@@ -150,22 +145,44 @@ private void cancelUpdate() {
 
     public void reloadHighlighter() {
         enableHighlighterAutoFormat();
-        highlightWithoutChange(getText());
+        highlightWithoutChange();
+    }
+
+    // This allows us to disable sending of accessibility events when highlighting is being
+    // performed. Sending of many accessibility events can cause performance issues and crashes.
+    // Blocking accessibility during _highlighting_ should have no impact on users.
+    @Override
+    public void sendAccessibilityEventUnchecked(AccessibilityEvent event) {
+        if (_accessibilityEnabled && getText().length() < 10000) {
+            super.sendAccessibilityEventUnchecked(event);
```

Comment:
@gsantner Added the fix for the accessibility / crash issue to this PR as there were some conflicting changes. I will close the other PR. This fix is a little better, but we still need to determine the correct max length and what the impact is on accessibility users.

Reply:
as far I can see accessbility is only disabled during highlighting runs, right? if so, should be fine

Reply:
Am still trying to figure out the minimal impact fix here. New version does 2 thing:
1. Block accessibility during highlighting
2. If the text in the accessibility message is too large, then cut it down

Reply:
This seems to be working so far. If we can do this properly, then there should be no impact on accessibility users while solving the crash issue